tiled/client/stream.py

```diff
@@ -0,0 +1,186 @@
+import inspect
+import threading
+import weakref
+from typing import Callable, List
+
+import anyio
+import httpx
+import msgpack
+from starlette.testclient import TestClient
+from websockets.sync.client import connect
+
+from tiled.client.context import Context
+
+Callback = Callable[["Subscription", dict], None]
+"A Callback will be called with the Subscription calling it and a dict with the update."
+
+
+class WebsocketWrapper:
```

Comment:
If this is implemented as _two_ wrapper classes, with identical public APIs, the check on `is_test_client` can be performed once (when you decide which wrapper class to instantiate, instead of internally on every call to `recv`.